auth_signup_verify_email/tests/test_verify_email.py

```diff
@@ -1,51 +1,32 @@
 # Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
+from lxml.html import document_fromstring
 from odoo import _
 from odoo.tests.common import HttpCase
 from odoo.tools.misc import mute_logger
-from lxml.html import document_fromstring
-
-
-class SignupVerifyEmailCase(HttpCase):
 
-    post_install = True
-    at_install = False
 
+class UICase(HttpCase):
     def setUp(self):
-        super(SignupVerifyEmailCase, self).setUp()
+        super(UICase, self).setUp()
+        with self.cursor() as cr:
+            env = self.env(cr)
```

Comment:
I don't get how that is different from this https://github.com/odoo/odoo/blob/11.0/odoo/tests/common.py#L180

Reply:
The difference is that it works 😆 

Well, now seriously, there's a big problem on documentation, and this took hours for me to guess, but basically:

- `HttpCase` uses a `TestCursor`, while all other tests use just a plain `Cursor`. This is because when you open any http call, it must be handled by a different thread, thus a different cursor too, and Odoo commits the transaction by itself always in such cases, and is multithreaded by default. The `TestCursor` converts the commits in savepoints and is a blocking cursor, so it circumvents all of this.
- [The test mode is enabled after the `self.env` is created](https://github.com/odoo/odoo/blob/4669a4a91fba1c8f731ebfbcfdef7833b9b305dc/odoo/tests/common.py#L261-L264), so actually you come to have one cursor in `self.env.cr` that is different from the cursor the http calls will use.
- If you use different cursors (i.e. creating some records with `self.env.cr` and then loading them through http), the changes between them will not be noticed, thus the errors.
- The test had a failure on committing. That's plain wrong in a test case, where *you should never commit*. Well, it rolled it back, but at the end it's worse than doing things the right way.
- When you do [`self.cursor()`](https://github.com/odoo/odoo/blob/4669a4a91fba1c8f731ebfbcfdef7833b9b305dc/odoo/tests/common.py#L119-L120) it instantiates the current cursor from the current registry, which basically means that you're now using the same `TestCursor` as your http calls. Now, inside that cursor, you can work as usual.

BTW, I pushed this to be on top of your branch, but it actually should be just commit aea1a71 and this one rebased on top of it. If you rebase as that, you'll see the diff is quite small from v10 😊

Reply:
I see, thanks for the full explanation :) 
I'm gonna rebase+cherry-pick your commit. tnx!